torch/fx/passes/utils/matcher_utils.py

```diff
@@ -146,9 +167,23 @@ def _match_nodes(self, pn: Node, gn: Node, match: InternalMatch) -> bool:
 
         # Recursively traverse upwards to check if `pn` is a true
         # match for `gn`
-        match_found = (len(pn.all_input_nodes) == len(gn.all_input_nodes) and
-                       all(self._match_nodes(pn_, gn_, match) for pn_, gn_
-                           in zip(pn.all_input_nodes, gn.all_input_nodes)))
+        match_found = True
+
+        pn_flatten_args, _ = pytree.tree_flatten(pn.args)
+        gn_flatten_args, _ = pytree.tree_flatten(gn.args)
+
+        if len(pn_flatten_args) == len(gn_flatten_args):
+            for pn_, gn_ in zip(pn_flatten_args, gn_flatten_args):
+                if isinstance(gn_, Node) and isinstance(pn_, Node):
+                    matched = self._match_nodes(pn_, gn_, match)
```

Comment:
nit: maybe we can just use `match_found` here

Reply:
In this context, matched and match_found carries different meaning... 

`matched` is for each argument, `match_found` is for current node
if and only if `matched` for all the argument is True, then `match_found` is True.